jl4-core/src/L4/Evaluate.hs

```diff
@@ -12,26 +21,39 @@ import L4.Utils.RevList
 
 import Data.Either
 
-newtype Eval a = MkEval (EvalState -> (Either EvalException a, EvalState))
-  deriving (Functor, Applicative, Monad, MonadError EvalException, MonadState EvalState)
-    via ExceptT EvalException (StateT EvalState Identity)
+newtype Eval a = MkEval (EvalState -> EvalEnv -> (Either EvalException a, EvalState))
```

Comment:
Still not extremely happy that we're adding a `Reader` component for just the URI, but ok ...